calc/vxlan_resolver.go

```diff
@@ -0,0 +1,476 @@
+		}
+
+		for _, r := range newRoutes {
+			logCxt := logrus.WithField("newRoute", r)
+			if currentRoutes.Contains(r) {
+				logCxt.Debug("Desired VXLAN route already exists, skip")
+				continue
+			}
+
+			c.blockToRoutes[key].Add(r)
+			adds.Add(r)
+		}
+
+		currentRoutes.Iter(func(item interface{}) error {
+			r := item.(vxlanRoute)
+
+			// For each existing route which is no longer present, we need to delete it.
+			if _, ok := newRoutes[r.Key()]; ok {
+				// Exists, and we want it to - nothing to do.
+				return nil
+			}
+
+			// Current route is not in new set - we need to withdraw the route, and also
+			// remove it from internal state.
+			deletes.Add(r)
+			c.blockToRoutes[key].Discard(r)
+			return nil
+		})
+
+		// At this point we've determined the correct diff to perform based on the block update.
+		// Delete any routes which are gone for good, withdraw modified routes, and send updates for
+		// new ones.
+		deletes.Iter(func(item interface{}) error {
+			c.withdrawRoute(item.(vxlanRoute))
+			return nil
+		})
+		c.kickPendingRoutes(adds)
+	} else {
+		// Block has been deleted. Clean up routes that were contributed by this block.
+		routes := c.blockToRoutes[key]
+		routes.Iter(func(item interface{}) error {
+			c.withdrawRoute(item.(vxlanRoute))
+			return nil
+		})
+		delete(c.blockToRoutes, key)
+	}
+	return
+}
+
+// OnHostIPUpdate gets called whenever a node IP address changes. On an add/update,
+// we need to check if there are VTEPs or routes which are now valid, and trigger programming
+// of them to the data plane. On a delete, we need to withdraw any routes and VTEPs associated
+// with the node.
+func (c *VXLANResolver) OnHostIPUpdate(update api.Update) (_ bool) {
+	nodeName := update.Key.(model.HostIPKey).Hostname
+	logCxt := logrus.WithField("node", nodeName)
+	pendingSet, sentSet := c.routeSets()
+	vtepSent := c.vtepSent(nodeName)
+	if update.Value != nil {
+		// Host IP updated or added. If it was added, we should check to see if we're ready
+		// to send a VTEP and associated routes. If we already knew about this one, we need to
+		// see if it has changed. If it has, we should remove and reprogram the VTEP and routes.
+		newIP := update.Value.(*net.IP).String()
+		currIP := c.nodeNameToIPAddr[nodeName]
+		logCxt = logCxt.WithFields(logrus.Fields{"newIP": newIP, "currIP": currIP})
+		if vtepSent {
+			if currIP == newIP {
+				// If we've already handled this node, there's nothing to do. Deduplicate.
+				logCxt.Debug("Skipping duplicate node IP update")
+				return
+			}
+
+			// We've already sent a VTEP for this node, and the node's IP address has changed.
+			// We need to revoke it and any routes before sending any updates.
+			logCxt.Info("Withdrawing routes and VTEP, node changed IP address")
+			sentSet.Iter(func(item interface{}) error {
+				r := item.(vxlanRoute)
+				if r.node == nodeName {
+					c.withdrawRoute(r)
+				}
+				return nil
+			})
+			c.sendVTEPRemove(nodeName)
```

Comment:
Yep. IIUC it will handle it properly the way it is currently written, will want to test though.

I think the current data plane will update its existing entries, triggering some updates routes in the route table.